python/cudf/cudf/core/buffer.py

```diff
@@ -104,6 +154,10 @@ def ptr(self) -> int:
     def size(self) -> int:
         return self._size
 
+    @property
+    def owner(self) -> object:
```

Comment:
Good point, I guess it is a matter of temperament.  

I think we want the type checker to warn us if we mistakenly assume the type of `.owner`. Since `Buffer` only uses `.owner` to keep the object alive, I would expect users to use `isinstance()` before assuming its type.  Or at least use `typing.cast()` to make the assumption explicit.

Now, we could define `owner` as a `Generic` type but I think that is taking it too far.

Reply:
I think if we want the type-checker to provide warnings, then the generic way is really the only option:
```
class Buffer(Generic[T]):
    def __init__(self, ..., owner: T):
        self.owner = owner
```
is not too bad. By annotating the return type as `object` we want to say "this could be anything" but the consequence for the type-checking is that we don't have any information (and everything gets underlined with squiggles). At least with `Any` you're saying "this could be anything" so you'd better check before you continue. Of course, generic types mean that you don't have that problem.

Reply:
Using `Any` now:
```python
class DeviceBufferLike(Protocol):
    @property
    def owner(self) -> Any:
        """Object owning the memory of the buffer."""
```